ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -98,184 +95,65 @@ import           Ouroboros.Consensus.Util.CBOR (decodeWithOrigin,
---   between cost of reapplying blocks and the cost of storing more snapshots.
---   The latter is primarily the cost of less opportunity for garbage
---   collection, which is hard to quantify abstractly. This should probably
---   be determined empirically.
---
--- Some example boundary cases:
---
--- * Suppose @k = 4@ and @snapEvery = 1@:
---
---   > *G  |> []               <| G
---   > *G  |> [L1]             <| L1
---   > *G  |> [L1, L2]         <| L2
---   > *G  |> [L1, L2, L3]     <| L3
---   > *G  |> [L1, L2, L3, L4] <| L4
---   > *L1 |> [L2, L3, L4, L5] <| L5
---   > *L2 |> [L3, L4, L5, L6] <| L6
---
---   Note that in this case the number of blocks will always be @k@ (unless we
---   are close to genesis), and the anchor is always the oldest point we can
---   roll back to.
---
--- * If @k = 1@ and @snapEvery = 4@, we get
---
---   > *G  |> [ ]                  <| G
---   > *G  |> [ B1]                <| L1
---   >  G  |> [*B1,  B2]           <| L2
---   >  G  |> [ B1, *B2,  B3]      <| L3
---   >  G  |> [ B1,  B2, *B3, L4]  <| L4
---   > *L4 |> [ B5]                <| L5
---   >  L4 |> [*B5,  B6]           <| L6
---   >  L4 |> [ B5, *B6,  B7]      <| L7
---   >  L4 |> [ B5,  B6, *B7, L8]  <| L8
---   > *L8 |> [*B9]                <| L9
---
---   (The minimum distance between the current ledger and the maximum rollback
---   is @k@.)
---
--- * If @k = 0@ and @snapEvery = 4@, we get
---
---   > *G  |> [ ]                  <| G
---   >  G  |> [*B1]                <| L1
---   >  G  |> [ B1, *B2]           <| L2
---   >  G  |> [ B1,  B2, *B3]      <| L3
---   > *L4 |> [ ]                  <| L4
---   >  L4 |> [*B5]                <| L5
---   >  L4 |> [ B5, *B6]           <| L6
---   >  L4 |> [ B5,  B6, *B7]      <| L7
---   > *L8 |> [ ]                  <| L8
---
---   @k = 0@ is the only case where the list might be empty. Of course, this is
---   not a particularly useful configuration.
---
--- * If @k = 1@ and @snapEvery = 1@, we get
---
---   > *G  |> [ ]   <| G
---   > *G  |> [ L1] <| L1
---   > *L1 |> [ L2] <| L2
---   > *L2 |> [ L3] <| L3
---
--- * If @k = 0@ and @snapEvery = 1@, we get
---
---   > *G  |> [ ] <| G
---   > *L1 |> [ ] <| L1
---   > *L2 |> [ ] <| L2
---
--- * Finally, if @k = snapEvery = k4@, we get
---
---   > *G  |> [ ]                             <| G
---   > *G  |> [ B1]                           <| L1
---   > *G  |> [ B1,  B2]                      <| L2
---   > *G  |> [ B1,  B2,  B3]                 <| L3
---   > *G  |> [ B1,  B2,  B3, L4]             <| L4
---   >  G  |> [*B1,  B2,  B3, L4, B5]         <| L5
---   >  G  |> [ B1, *B2,  B3, L4, B5, B6]     <| L6
---   >  G  |> [ B1,  B2, *B3, L4, B5, B6, B7] <| L7
---   > *L4 |> [ B5,  B6,  B7, L8]             <| L8
---   >  L4 |> [*B5,  B6,  B7, L8, B9]         <| L9
+-- > L3 |> []
 data LedgerDB l r = LedgerDB {
-      -- | The ledger state at the tip of the chain
-      ledgerDbCurrent :: !l
-
       -- | Older ledger states
```

Comment:
Now that `ledgerDbCurrent` is no longer a field, we only have these "older ledger states", which includes the current one. So rename it to:
```suggestion
      -- | Ledger state checkpoints
```